tests/put.rs

```diff
@@ -0,0 +1,487 @@
+    let cmd = c.args([
+        "--region",
+        "local",
+        "--table",
+        &table_name,
+        "put",
+        "42",
+        "abc",
+        "--item",
+        r#"{"a": 9, "b": "str"}"#,
+    ]);
+    cmd.assert().success().stdout(format!(
+        "Successfully put an item to the table '{}'.\n",
+        table_name
+    ));
+
+    let mut c = tm.command()?;
+    let get_cmd = c.args([
+        "--region",
+        "local",
+        "--table",
+        &table_name,
+        "scan",
+        "-o",
+        "raw",
+    ]);
+
+    let expected = r#"
+      [{
+          "pk": {
+              "S": "42"
+          },
+          "sk": {
+              "S": "abc"
+          },
+          "a": {
+              "N": "9"
+          },
+          "b": {
+              "S": "str"
+          }
+      }]
+      "#;
+
+    util::assert_eq_json_ignore_order(get_cmd, expected);
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_put_complex_item() -> Result<(), Box<dyn std::error::Error>> {
+    let mut tm = util::setup().await?;
+    let table_name = tm.create_temporary_table("pk", None).await?;
+
+    let mut c = tm.command()?;
+    let cmd = c.args([
+        "--region",
+        "local",
+        "--table",
+        &table_name,
+        "put",
+        "42",
+        "--item",
+        r#"{"myfield": "is", "nested": {"can": true, "go": false, "deep": [1,2,{"this_is_set": <<"x","y","z">>}]}}"#,
+    ]);
+    cmd.assert().success().stdout(format!(
+        "Successfully put an item to the table '{}'.\n",
+        table_name
+    ));
+
+    let mut c = tm.command()?;
+    let get_cmd = c.args([
+        "--region",
+        "local",
+        "--table",
+        &table_name,
+        "scan",
+        "-o",
+        "raw",
+    ]);
+
+    let expected = r#"
+        [{
+          "pk": {
+              "S": "42"
+          },
+          "myfield": {
+              "S": "is"
+          },
+          "nested": {
+              "M": {
+                  "can": {
+                      "BOOL": true
+                  },
+                  "deep": {
+                      "L": [{
+                          "N": "1"
+                      }, {
+                          "N": "2"
+                      }, {
+                          "M": {
+                              "this_is_set": {
+                                  "SS": ["x", "y", "z"]
+                              }
+                          }
+                      }]
+                  },
+                  "go": {
+                      "BOOL": false
+                  }
+              }
+          }
+      }]
+      "#;
+
+    util::assert_eq_json_ignore_order(get_cmd, expected);
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_put_same_pk() -> Result<(), Box<dyn std::error::Error>> {
+    let mut tm = util::setup().await?;
+    let table_name = tm
+        .create_temporary_table_with_items(
+            "pk",
+            None,
+            vec![util::TemporaryItem::new(
```

Comment:
Actually, we no longer do not need to use vec!. We can just pass the array. However, we can modify it later in other pull request.

Reply:
I think this should be handled in another issue based on current usage of the method in other tests.